### ISSUE TITLE
[1.16.5] Behavior of model rotation using Forge's TRSR transformation extension to vanilla model seems confusing

### ISSUE THREAD

**Issue by 3TUSK** [MinecraftForge/MinecraftForge#8069]
**Minecraft Version:** 1.16.5

**Forge Version:** 36.2.0

**Logs:** N/A, there isn't a crash

**Steps to Reproduce:**
 1. Download this test mod workspace: [test-kit.zip](https://github.com/MinecraftForge/MinecraftForge/files/7118763/test-kit.zip)
 2. Setup, then run its data-gen to get all models generated
 3. Launch the game, create a new world
 4. `/setblock ~ ~ ~ rotate_test:x_15` and observe the model. There are also `x_30`, `x_45`, … `x_165`, `y_15`, `y_30`, … `z_15`, `z_30`, … which can be tested simultaneously.

**Description of issue:**

If I specify a model like this for my block:

```json
{
  "parent": "minecraft:block/oak_planks",
  "transform": {
    "rotation": [ 15, 0, 0 ]
  }
}
```

The result model is not the *model* itself rotating 15 degrees around X-axis, with default origin. It actually looks like the following (the second one from left):
![2021-09-06_20 01 41](https://user-images.githubusercontent.com/9113033/132282073-a036fe08-336f-4b4d-826a-7e0a2cb116d0.png)

Notably, the block on the right-most looks correct: the model itself is indeed rotated 90 degrees with origin of `opposing-corner` around X-axis. Bottom side is culled because the block is placed on top of the white concrete wall.

With a closer look and selection box:
![2021-09-07_00 43 56](https://user-images.githubusercontent.com/9113033/132305163-d687dcfd-9526-4469-97c5-95431710edd1.png)
The model looks more like "stretched" instead of "rotated". 

Further tests shows that if the rotation is not in multiple of 90 degrees (regardless which axises), the model will looks more like "stretched" rather than "rotated". The test mod provided above include examples for this.

Given that there aren't documentations about this feature other than Forge's own test mods (if those count), I am not certain if this is the expected behavior.

**Comment by gigaherz** [comment]
Due to a mistake in the initial implementation, the default origin for transformations was set to be at a corner... but not the usual corner (0,0,0). This was due to a misunderstanding of the original code: instead of moving the origin from the corner to the center, I moved it from the center to the opposite corner.
Since the bug was discovered long after forge had a stable release with the feature, it was decided that we shouldn't change the semantics of the file format without a clear spec bump that would indicate older files need to be looked at.

**Comment by 3TUSK** [comment]
I am aware about the issue of wrong corner; however, that is not the issue that I am trying to express here; plus, even though the corner is wrong, the rotation is still right (specifically, the one with 90 degrees rotation) in terms of that rotation origin. 

See my updated screenshots for a closer look of the "seemingly wrong" model. The problem is that, rotation with non-multiple-of-90 degrees will result in a "stretched" model rather than "rotated" model.

**Comment by gigaherz** [comment]
Oh I see, I was confused what the picture was trying to show, didn't realize what it was.
I have no idea why that would happen. It worked fine last time I tested it, although that may have been back in 1.15.

**Comment by 3TUSK** [comment]
> although that may have been back in 1.15

FWIW, the said issue can also be reproduced in 1.15.2+Forge 31.2.55. The following image shows an oak plank model with rotation of 15 degree around x-axis (supposed). Could this issue date back to even earlier version?
![2021-09-07_13 31 50](https://user-images.githubusercontent.com/9113033/132407214-21e29987-3626-46fb-92b5-edbf8ba36ab2.png)

**Comment by gigaherz** [comment]
It could. I'm now wondering if I ever tried it with a vanilla block model ("elements" model). I'm suspecting I only tried with item or OBJ models... Will have to investigate, when I have the time.

**Comment by XFactHD** [comment]
This issue is fixed in 1.19.4, most likely by virtue of #9410 (backported to 1.19.2 by #9411), tested with the test mod provided above with the bare minimum changes to make it compile and run on 1.19.4:
![2023-05-15_01 11 44](https://github.com/MinecraftForge/MinecraftForge/assets/11262040/8d71047f-169b-41a9-95f0-9e323902e63b)

**Comment by SizableShrimp** [comment]
As this appears infeasible to fix in 1.18 (see [context in Discord](https://discord.com/channels/313125603924639766/852298000042164244/1107457731998265394)) and 1.16 is out of support, this issue is resolved.
